[DNM] [WIP] Update pairing and bls crates

vihu opened the pull request:
This is gonna take a while and lots of alcohol

vihu committed:
Take wild shots at bumping pairing and bls

JayKickliter commented on src/lib.rs:
```diff
@@ -58,15 +56,6 @@ use util::sha3_256;
 #[cfg(feature = "use-insecure-test-only-mock-crypto")]
 mod mock;
 
-#[cfg(feature = "use-insecure-test-only-mock-crypto")]
```
are you deleting this altogether with no plan of replacing it? Fine with me. Nothing better than deleted code

vihu replied:
I think I deleted this in rage, I'll probably add it back at some point (if need be)

vihu committed:
Try to make sense of things

vihu committed:
More broken WIP

- Update group dependency
- Update lib.rs
- Update poly.rs
- Update serde_impl
- Remove cmp_pairing

vihu committed:
More things, more brokenness

vihu committed:
Lesser errors, no less pain

vihu committed:
Whoops

vihu committed:
Remove serde_impl entirely and try other things

vihu committed:
Still thoroughly broken but making progress

vihu committed:
Fix clear_fr

vihu committed:
Delete Fr, back to 100+ errors

vihu committed:
Errors are down to 82 only

vihu committed:
More compiler error reduction

vihu committed:
Yolo
